quil-rs/src/expression.rs

```diff
@@ -253,6 +253,537 @@ fn is_small(x: f64) -> bool {
+                                expression,
+                            })),
+                            "sqrt" => Ok(Expression::FunctionCall(FunctionCallExpression {
+                                function: ExpressionFunction::SquareRoot,
+                                expression,
+                            })),
+                            "pos" => Ok(Expression::Prefix(PrefixExpression {
+                                operator: PrefixOperator::Plus,
+                                expression,
+                            })),
+                            "neg" => Ok(Expression::Prefix(PrefixExpression {
+                                operator: PrefixOperator::Minus,
+                                expression,
+                            })),
+                            _ => Err(SimplificationError::UnexpectedUnaryExpr(s.to_string())),
+                        }
+                    }
+                    [Sexp::String(s), Sexp::String(name), Sexp::String(y)]
+                        if s.as_str() == "address" =>
+                    {
+                        let index = y
+                            .parse::<u64>()
+                            .map_err(SimplificationError::IndexExpected)?;
+                        Ok(Expression::Address(MemoryReference {
+                            name: name.clone(),
+                            index,
+                        }))
+                    }
+                    [Sexp::String(s), x, y] => {
+                        let left = Expression::try_from(x)?.into();
+                        let right = Expression::try_from(y)?.into();
+                        let operator = match s.as_str() {
+                            "^" => InfixOperator::Caret,
+                            "*" => InfixOperator::Star,
+                            "/" => InfixOperator::Slash,
+                            "+" => InfixOperator::Plus,
+                            "-" => InfixOperator::Minus,
+                            _ => return Err(SimplificationError::UnexpectedInfixOp(s.to_string())),
+                        };
+                        Ok(Expression::Infix(InfixExpression {
+                            left,
+                            right,
+                            operator,
+                        }))
+                    }
+                    _ => Err(SimplificationError::UnexpectedListSexp(sexp.clone())),
+                },
+            }
+        }
+    }
+
+    /// An [`egg`]-friendly complex number.
+    /// We can't jsut use `num_complex::Complex64`, because we need `Ord` and `Hash`.
+    ///
+    /// Fun fact, there is no total ordering on the complex numbers; however, the derived thing
+    /// here will work for our purposes.
+    ///
+    /// https://en.wikipedia.org/wiki/Complex_number#Ordering
+    #[derive(Debug, Default, PartialEq, Eq, PartialOrd, Ord, Clone, Copy, Hash)]
+    struct C {
+        re: OrderedFloat<f64>,
+        im: OrderedFloat<f64>,
+    }
```

Comment:
Would prefer a different name for this -- I looked at `impl C` below and got confused at why you were trying to add C bindings (mistook it for `extern "C"`)